packages/teleport/src/Nodes/AddNode/Iam/Iam.tsx

```diff
@@ -0,0 +1,267 @@
+import React from 'react';
+import styled from 'styled-components';
+import { DialogContent, DialogFooter } from 'design/Dialog';
+import {
+  Text,
+  Box,
+  ButtonPrimary,
+  Link,
+  Alert,
+  ButtonLink,
+  ButtonSecondary,
+} from 'design';
+import TextSelectCopy from 'teleport/components/TextSelectCopy';
+import FieldInput from 'shared/components/FieldInput';
+import Validation, { Validator } from 'shared/components/Validation';
+import { Attempt } from 'shared/hooks/useAttemptNext';
+import { Rule } from 'teleport/services/joinToken';
+import DownloadLinks from 'teleport/components/DownloadLinks';
+import { Pencil } from 'design/Icon';
+
+export default function Iam({
+  token,
+  expiry,
+  attempt,
+  onGenerate,
+  isEnterprise,
+  version,
+  onClose,
+}: Props) {
+  const { hostname, port } = window.document.location;
+  const host = `${hostname}:${port || '443'}`;
+
+  const [rules, setRules] = React.useState<RuleForm[]>([
+    {
+      awsAccount: '',
+      awsArn: '',
+      isCollapsed: false,
+    },
+  ]);
+
+  function setRuleAtIndex(index: number, rule: RuleForm) {
+    const newRules = [...rules];
+    newRules[index] = rule;
+    setRules(newRules);
+  }
+
+  function handleRemove(index: number) {
+    const newRules = [...rules];
+    newRules.splice(index, 1);
+    setRules(newRules);
+  }
+
+  function handleGenerate(validator: Validator) {
+    if (!validator.validate()) {
+      return;
+    }
+
+    setRules(rules.map((rule: RuleForm) => ({ ...rule, isCollapsed: true })));
+
+    onGenerate(rules);
+  }
+
+  return (
+    <Validation>
+      {({ validator }) => (
+        <>
+          <DialogContent flex="0 0 auto">
+            {attempt.status === 'failed' && (
+              <Alert kind="danger" children={attempt.statusText} />
+            )}
+            <Text mb={4}>
+              {`Add any Teleport Node or Proxy
+              with access to IAM credentials. `}
+              {`You can read more
+              about the IAM method in the `}
+              <Link href="https://goteleport.com/docs/setup/guides/joining-nodes-aws/">
+                documentation
+              </Link>
+              .
+            </Text>
+            <Box mb={4}>
+              <Text bold as="span">
+                Step 1
+              </Text>{' '}
+              - Download Teleport package to your computer
+              <DownloadLinks isEnterprise={isEnterprise} version={version} />
+            </Box>
+            <Box>
+              <Text bold as="span">
+                Step 2
+              </Text>{' '}
+              - Setup AWS Rules
+              {rules.map((rule, index) => (
+                <RuleBox
+                  key={index}
+                  bg="primary.lighter"
+                  borderRadius="2"
+                  mt={1}
+                  p="3"
+                  position="relative"
+                >
+                  <Text typography="h5">{`Rule #${index + 1}`}</Text>
```

Comment:
i suggest `h6`, i think smaller or equal text size for inner content flows better visually